server/src/main/java/org/elasticsearch/index/codec/vectors/diskbbq/next/ESNextDiskBBQVectorsReader.java

```diff
@@ -512,23 +512,32 @@ private static void score(
             }
         }
 
-        for (; i < size; i++) {
-            int centroidOrd = scoresOffset + i;
-            if (acceptCentroids == null || acceptCentroids.get(centroidOrd)) {
-                float score = scorer.score(
+        int tailBulkSize = size - i;
+        if (tailBulkSize > 0) {
+            if (acceptCentroids == null || acceptCentroids.cardinality(scoresOffset + i, scoresOffset + i + tailBulkSize) > 0) {
+                float[] tailScores = new float[tailBulkSize];
```

Comment:
Cannot we reuse the scores array passed in this method?

Reply:
good catch 👍

Reply:
addressed in https://github.com/elastic/elasticsearch/pull/139835/commits/c4d90e260c793baa01f8c065457cf569c70ee93f